src/provider/provider_ctl_stats_impl.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2025 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_PROVIDER_CTL_STATS_IMPL_H
+#define UMF_PROVIDER_CTL_STATS_IMPL_H 1
+#else
+#error __FILE__"should not be included more than once"
+#endif
+
+#ifndef CTL_PROVIDER_TYPE
+#error "CTL_PROVIDER_TYPE must be defined"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ctl/ctl.h"
+#include <utils/utils_assert.h>
+
+static int CTL_READ_HANDLER(peak_memory)(void *ctx,
+                                         umf_ctl_query_source_t source,
+                                         void *arg,
+                                         umf_ctl_index_utlist_t *indexes,
+                                         const char *extra_name,
+                                         umf_ctl_query_type_t query_type) {
+    /* suppress unused-parameter errors */
+    (void)source, (void)indexes, (void)ctx, (void)extra_name, (void)query_type;
+
+    size_t *arg_out = arg;
+    CTL_PROVIDER_TYPE *provider = (CTL_PROVIDER_TYPE *)ctx;
+    COMPILE_ERROR_ON(sizeof(provider->stats.peak_memory) != sizeof(uint64_t));
+    utils_atomic_load_acquire_u64((uint64_t *)&provider->stats.peak_memory,
```

Comment:
why there are casts to (uint64_t) here? is peak_mem already defined as uint64_t?

Reply:
Done